services/barcodereader.md

```diff
@@ -0,0 +1,43 @@
+# Barcode reader
+
+    identifier: 0x1c739e6c
+
+A device that reads various barcodes, like QR codes. For the web, see [BarcodeDetector](https://developer.mozilla.org/en-US/docs/Web/API/BarcodeDetector).
+
+## Registers
+
+    rw enabled: bool @ intensity
+    
+Turns on or off the detection of barcodes.
+
+    enum Format: u32 {
+        Aztec = 1
+        Code128 = 2
+        Code39 = 3
+        Code93 = 4
+        Codabar = 5
+        DataMatrix = 6
+        Ean13 = 8
+        Ean8 = 9
+        ITF = 10
+        Pdf417 = 11
+        QrCode = 12
+        UpcA = 13
+        UpcE = 14
+    }
+    const formats? @ 0x80 {
+        repeats:
+            format: Format;  
+    }
+    
+Reports the list of supported barcode formats, as documented in https://developer.mozilla.org/en-US/docs/Web/API/Barcode_Detection_API.
+
+## Events
+
+    event detect @ active {
+        format: Format
+        data: bytes 
```

Comment:
```suggestion
        data: string
```

the webapi uses strings, and I think none of these barcodes are binary
it also makes it clear that EAN13 should read as `"088381463935"` and not say bytes `0x00, 0x08, 0x08, ...`

Reply:
Good catch. Web api is strong too